### PR TITLE
Add mutations for ItemLists

### PR DESCRIPTION
**Pull request by alastair**

### CODE REVIEW

File: trompace/mutations/itemlist.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        identifier: The unique identifier of the listitem object.
+
+    Returns:
+        The string for the mutation for deleting the listitem object based
+        on the identifier.
+    """
+    return format_mutation("DeleteListItem", {"identifier": identifier})
+
+
+def mutation_add_itemlist_itemlist_element(identifier_1: str,
```
Comment (alastair) [inline trompace/mutations/itemlist.py:162]: In some places where these methods can be used for multiple different types, I used `identifier_1` and `identifier_2`. However, in the case where we always know what the type is, we can use a more descriptive name.
In the case of this method, the first argument is always an itemlist identifier, so lets call it `itemlist_id` instead.

Remember that an ItemListElement can be any ThingInterface, so the documentation should be updated to indicate that this is possible

File: trompace/mutations/itemlist.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        contributor: A person, an organization, or a service responsible
+        for contributing the ItemList to the web resource.
+        This can be either a name or a base URL.
+        description: The description of the ItemList object
+        """
+
+
+@docstring_interpolate("listitem_args", LISTITEM_ARGS_DOCS)
+def mutation_create_listitem(contributor: str, name: str = None,
+                             description: str = None):
```
Comment (alastair) [inline trompace/mutations/itemlist.py:103]: did we discuss that we would add position here too? I think it's a simple addition to put in.

**Review by alastair — COMMENTED** [review]
This looks great, thanks for taking the time to work on it.

In addition to the comments, there are two more things that I'd like to see here:

 1. Mutations for setting/removing ListItem.nextItem (`MergeListItemNextItem`)

 2. From the specification of the schema, an itemListElement can be any Thing. This means that in your case, if you just want an unordered list of music recordings, you can just attach the recordings directly to the ItemList, e.g. `mutation_add_itemlist_itemlist_element(list_id, recording_id)`
 However, in the case that someone wants an ordered list, we need to add an intermediate ListItem, so we'd have 
> ItemList -> itemListElement -> ListItem (with position set) -> item -> MusicRecording

Because of this, I think that we should also add mutations for `MergeListItemItem` (and RemoveListItemItem)

File: trompace/constants.py [hunk 3]
```diff
@@ -14,3 +14,13 @@ class ActionStatusType(enum.Enum):
 
     def __str__(self):
         return self.name
+
+
+class ItemListOrderType(enum.Enum):
+    ascending = enum.auto()
```
[the remark this answers is not in the document]
Comment (LPorcaro) [reply trompace/constants.py:20]: Any source from where checking how to 

> add some documentation here to say that this is based on the schema.org type (just a link to the URL is enough)

File: trompace/mutations/itemlist.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        description: The description of the ItemList object
+        """
+
+
+@docstring_interpolate("itemlist_args", ITEMLIST_ARGS_DOCS)
+def mutation_create_itemlist(contributor: str, name: str = None,
+                             itemlistorder: ItemListOrderType =
+                             ItemListOrderType.unordered,
+                             description: str = None):
+    """Returns a mutation for creating an itemlist object.
```
[the remark this answers is not in the document]
Comment (LPorcaro) [reply trompace/mutations/itemlist.py:22]: Also in this case, how do I have to " link to the schema.org page for the type."? (in the documentation I suppose)

[the remark this answers is not in the document]
Comment (alastair) [reply trompace/mutations/itemlist.py:22]: I have a small example here: https://github.com/trompamusic/trompa-ce-client/blob/master/trompace/mutations/mediaobject.py#L157-L159

I notice that I've not been very consistent in doing this, but I think that it's a good idea to start adding it when our objects match directly to schema.org objects

File: trompace/constants.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (alastair) [reply trompace/constants.py:20]: I would just add documentation to the class saying:
```
"""This represents a https://schema.org/ItemListOrderType"""
```

**Pull request merged**